routes/post/post_program.js

```diff
@@ -0,0 +1,51 @@
+'use strict';
+const _ = require('underscore');
+const keystone = require('keystone');
+const Program = keystone.list('Program');
+
+exports = module.exports = (req, res) => {
+	const locals = res.locals;
+	const action = req.body.action;
+	
+	function fillProgram(program, editing) {
+		const updater = program.getUpdateHandler(req, res, {
+			errorMessage: 'There was an error creating your new program:'
+		});
+		return new Promise((resolve, reject) => {
+			updater.process(req.body, {
+				flashErrors: true,
+				logErrors: true,
+				fields: 'title, djs, schedule, isBiweekly, description, genre'
+			}, err => {
+				if (err) {
+					return reject(err);
+				} else {
+					req.flash('success', 'Program ' + editing ? 'updated' : 'created');
+					return resolve(program);
+				}
+			});
+		});
+	}
+	
+	if (action === 'add-program') {
```

Comment:
I'm not able to test it right now, but does this properly reflect what happens when you add a program with a scheduling conflict? Right now, the DB sends an error to the callback/promise and prevents the program from being saved. I'm not sure what happens when you supply 2 arrow functions as I haven't used promises much.